integration/k8s_fixture_test.go

```diff
@@ -64,31 +64,60 @@ func (f *k8sFixture) CurlUntil(ctx context.Context, url string, expectedContents
 	}, expectedContents)
 }
 
-// Waits until all pods matching the selector are ready (i.e. phase = "Running")
+// Waits until n pods matching the selector are ready (i.e. phase = "Running").
+// If n = -1, waits for all pods matching the selector to be ready
 // At least one pod must match.
 // Returns the names of the ready pods.
+func (f *k8sFixture) WaitForPodsReady(ctx context.Context, selector string, n int) []string {
+	return f.WaitForPodsInPhase(ctx, selector, v1.PodRunning, n)
+}
+
 func (f *k8sFixture) WaitForAllPodsReady(ctx context.Context, selector string) []string {
-	return f.WaitForAllPodsInPhase(ctx, selector, v1.PodRunning)
+	return f.WaitForPodsInPhase(ctx, selector, v1.PodRunning, -1)
 }
 
-func (f *k8sFixture) WaitForAllPodsInPhase(ctx context.Context, selector string, phase v1.PodPhase) []string {
+func (f *k8sFixture) WaitForOnePodWithAllContainersReady(ctx context.Context, selector string, timeout time.Duration) string {
```

Comment:
This new `n` parameter adds a bunch of complexity, but I couldn't find any justification for it. What's wrong with waiting on all pods? Can you leave some comments on what this is for?

Reply:
it was to address a blip i was seeing locally (where I'd get multiple ready pods back, presumably from catching the crash rebuild at exactly the wrong time) -- but I can leave this out unless other people start seeing that blip/it starts happening on CI